core/src/test/java/nl/nn/adapterframework/pipes/SoapWrapperPipeTest.java

```diff
@@ -34,41 +41,119 @@ public void testUnwrap() throws Exception {
 		pipe.configure();
 		pipe.start();
 
-		String input = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\"><soapenv:Body>"
-				+"<root xmlns=\""+TARGET_NAMESPACE+"\">\n"
-				+"<attrib>1</attrib>\n"
-				+"<attrib>2</attrib>\n"
-				+"</root></soapenv:Body></soapenv:Envelope>";
-		String expected = "<root xmlns=\""+TARGET_NAMESPACE+"\">\n"
-				+"<attrib>1</attrib>\n"
-				+"<attrib>2</attrib>\n"
-				+"</root>";
+		String input = "<soapenv:Envelope xmlns:soapenv=\"" + SOAPConstants.URI_NS_SOAP_1_1_ENVELOPE + "\"><soapenv:Body>"
+				+ DEFAULT_BODY_END;
+		String expected = "<root xmlns=\"" + TARGET_NAMESPACE + "\">\n"
+				+ "<attrib>1</attrib>\n"
+				+ "<attrib>2</attrib>\n"
+				+ "</root>";
 
-		PipeRunResult prr = doPipe(pipe, input,new PipeLineSession());
+		PipeRunResult prr = doPipe(pipe, input, new PipeLineSession());
 
 		String actual = prr.getResult().asString();
 
 		TestAssertions.assertEqualsIgnoreCRLF(expected, actual);
 	}
 
+	@Test
+	public void testShouldKeepPipeSoapConfigurationVersionWhileWrapping() throws Exception {
+		pipe.setDirection(Direction.UNWRAP);
+		pipe.registerForward(new PipeForward("pipe2", "READY"));
+		pipe.setStoreResultInSessionKey("originalMessage_unwrapped");
+		pipe.setRemoveOutputNamespaces(true);
+		pipe.configure();
+		pipe.start();
+
+		P wrapPipeSoap11 = createPipe();
+		wrapPipeSoap11.setDirection(Direction.WRAP);
+		wrapPipeSoap11.setSoapVersion(SoapVersion.SOAP11);
+		autowireByType(wrapPipeSoap11);
+		wrapPipeSoap11.registerForward(new PipeForward("success", "READY"));
+		wrapPipeSoap11.setName("pipe2");
+		pipeline.addPipe(wrapPipeSoap11);
+		wrapPipeSoap11.configure();
+		wrapPipeSoap11.start();
+
+		// Arrange 1
+		String inputSoap12 = "<soapenv:Envelope xmlns:soapenv=\"" + SOAPConstants.URI_NS_SOAP_1_2_ENVELOPE + "\"><soapenv:Body>" + DEFAULT_BODY_END;
+
+		PipeLineSession pipeLineSession = new PipeLineSession();
+		// Act
+		PipeRunResult prr = doPipe(pipe, inputSoap12, pipeLineSession);
+		PipeRunResult pipeRunResult = doPipe(wrapPipeSoap11, prr.getResult(), pipeLineSession);
```

Comment:
Ik snap niet goed wat hier gebeurd. Je krijgt `soap 1.2`, en geeft `soap 1.1` terug?
Op zich prima, de 2e pipe zet immers de `setSoapVersion` op `1.1`.

Vervolgens doe je alles opnieuw maar dan stuur, en return je `soap 1.1`.
Hoe vertaalt dit zich naar `ShouldKeepPipeSoapConfigurationVersionWhileWrapping` Dat hij altijd `soap 1.1` moet gebruiken?

Reply:
Ah ja, inderdaad is het logischer om het dan andersom te testen. Heb het nu aangepast, dus:
- Check A) SOAP 1.2 bericht er in -> unwrap -> wrap pipe=1.1 -> assert 1.1
- Check B) SOAP 1.1 bericht er in -> unwrap -> wrap pipe=1.2 -> assert 1.2